crates/ty_server/src/server.rs

```diff
@@ -47,23 +49,40 @@ impl Server {
         in_test: bool,
     ) -> crate::Result<Self> {
         let (id, init_value) = connection.initialize_start()?;
-        let init_params: InitializeParams = serde_json::from_value(init_value)?;
-
-        let AllOptions {
-            global: global_options,
-            workspace: mut workspace_options,
-        } = AllOptions::from_value(
-            init_params
-                .initialization_options
-                .unwrap_or_else(|| serde_json::Value::Object(serde_json::Map::default())),
-        );
 
-        let client_capabilities = init_params.capabilities;
+        let InitializeParams {
+            initialization_options,
+            capabilities: client_capabilities,
+            workspace_folders,
+            ..
+        } = serde_json::from_value(init_value)?;
+
+        let (initialization_options, deserialization_error) =
+            InitializationOptions::from_value(initialization_options);
+
+        if !in_test {
+            crate::logging::init_logging(
+                initialization_options.log_level.unwrap_or_default(),
+                initialization_options.log_file.as_deref(),
+            );
+        }
+
+        tracing::info!("Initialization options: {initialization_options:?}");
```

Comment:
I'd change this to debug or use a proper `Display` for `initialization_options`. `info` level messages are meant for users.

Reply:
Yeah, I've converted to `debug` for now but it might be useful to have the `display_settings` macro available by moving it out of `ruff_linter`.